spec.md

```diff
@@ -0,0 +1,948 @@
+In its absence implementations SHOULD default to the start of the medley section (if available).
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `PREVIEWSTART` was specified in seconds as a floating point number.
+> Version 2.0.0 of this specification changes this to an integer and milliseconds.
+
+### 3.14. The `MEDLEYSTART` and `MEDLEYEND` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        2.0.0
+```
+
+The `MEDLEYSTART` and `MEDLEYEND` headers indicate in milliseconds the start and end of the medley section of a song relative to the start of the audio.
+These tags replace `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` in version 2.0.0 of this specification.
+
+### 3.15. The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+Removed:      2.0.0
+```
+
+The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` headers indicate in beats the start and end of the medley section of a song.
+Implementations MUST respect the `GAP` value when calculating the medley start and end times.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> The headers `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` have been replaced by `MEDLEYSTART` and `MEDLEYEND` in version 2.0.0 of this specification.
+
+### 3.16. The `CALCMEDLEY` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       "on" / "off"
+Since:        0.2.0
+```
+
+If `MEDLEYSTART` or `MEDLEYEND` (`MEDLEYSTARTBEAT` or `MEDLEYENDBEAT` for versions before 2.0.0) are not specified,
+the `CALCMEDLEY` header indicates whether an implementation is supposed to determine the start and end of the medley section automatically.
+The value of this header is compared case-insensitively.
+
+> [!CAUTION]
+>
+> The exact semantics of the `CALCMEDLEY` header have not been defined yet.
+
+### 3.17. The `TITLE` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `TITLE` header contains the title of the song.
+
+### 3.18. The `ARTIST` Header
+
+```
+Required:     Yes
+Multi-Valued: No
+Since:        0.1.0
+```
+
+The `ARTIST` header contains the artist of the song.
+
+### 3.19. The `YEAR` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       4DIGIT
+Since:        0.2.0
+```
+
+The `YEAR` indicates the year in which the song was released.
+The value must be a positive integer.
+
+### 3.20. The `GENRE` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `GENRE` defines the genre(s) of the song.
+Individual genre values MUST be compared case-insensitively.
+For consistency, it is usually best to capitalize genres.
+
+> [!CAUTION]
+>
+> Whether genres should be compared case-insensitively or not hasn't been decided yet.
+
+### 3.21. The `LANGUAGE` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `LANGUAGE` header indicates the spoken or sung language(s) of a song.
```

Comment:
Can we refer to an iso standardization of English language names?

Reply:
I'd love for the values of this header to be standardized. I included this and a reference to [ISO 639-2](https://www.loc.gov/standards/iso639-2/php/code_list.php) which seems to be an appropriate standard.

Depending on the target audience of a file (humans or programs) it might even make sense to restrict values to language tags according to [RFC 4646](https://www.rfc-editor.org/rfc/rfc4646.html). This would have the added benefit of being able to differentiate American English (`en-US`) from British English (`en-GB`). I think this is best discussed in its own issue.

Reply:
Not all songs use real-world languages. Also, this attribute is currently used to indicate the writing system as well, e.g. 'Japanese (romanized)'.

Reply:
That's a very good point I hadn't thought of. Seems like [ISO 15924](https://en.wikipedia.org/wiki/ISO_15924) would be a standardization of writing systems. But considering the use of fictional languages (some of which are actually included in ISO 639-2) maybe it's better to leave it at a recommendation that when referring to languages, the English language name should be used for consistency.